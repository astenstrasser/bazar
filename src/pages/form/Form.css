.form-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 69vh;
}

.form {
  min-height: 30vh;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
}

@media (max-width: 600px) {
  .form {
    margin-top: 1rem;
  }
}

.form-field {
  margin-top: 1rem;
}

.form-field input {
  margin-left: 0.7rem;
  padding: 0.3rem;
}

.submit {
  margin-top: 1rem;
  padding: 0.4rem 1.5rem;
}
