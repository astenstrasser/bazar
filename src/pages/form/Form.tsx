import React, { useEffect, useState } from "react";
import "./Form.css";
import ReturnIcon from "../../shared-components/return-icon/ReturnIcon";

interface FormProps {
  handleSubmit: (name: string, price: number) => void;
}

function Form({ handleSubmit }: FormProps): React.ReactElement {
  const [isValid, setIsValid] = useState<boolean>(false);
  const [name, setName] = useState<string>("");
  const [price, setPrice] = useState<number>(0);

  useEffect(() => {
    price > 0 && !!name ? setIsValid(true) : setIsValid(false);
  }, [price, name]);

  return (
    <>
      <div className="form-container">
        <div className="form">
          <div className="form-field">
            <label htmlFor="name">Nome do Produto:</label>
            <input
              id="name"
              type="text"
              value={name}
              onChange={(event) => setName(event.target.value)}
            />
          </div>

          <div className="form-field">
            <label htmlFor="price">Pre√ßo:</label>
            <input
              id="price"
              value={price}
              type="number"
              onChange={(event) => setPrice(Number(event.target.value))}
            />
          </div>

          <button
            className="submit"
            onClick={() => handleSubmit(name, price)}
            disabled={!isValid}
          >
            Submit
          </button>
        </div>
      </div>
      <ReturnIcon reference={"/"} />
    </>
  );
}

export default Form;
