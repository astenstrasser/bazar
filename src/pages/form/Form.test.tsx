import React from "react";
import Form from "./Form";
import { render } from "@testing-library/react";

test("renders snapshot", () => {
  const { container } = render(<Form />);
  expect(container).toMatchSnapshot();
});
