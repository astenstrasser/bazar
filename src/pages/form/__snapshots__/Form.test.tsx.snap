// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<div>
  <div
    class="form-container"
  >
    <form>
      <label
        class="form-field"
      >
        Nome do Produto:
        <input
          class="form-field--input"
          name="name"
          type="text"
          value=""
        />
      </label>
      <label
        class="form-field"
      >
        Preço:
        <input
          class="form-field--input"
          name="price"
          type="number"
          value="0"
        />
      </label>
      <input
        class="submit"
        disabled=""
        type="submit"
        value="Enviar"
      />
    </form>
    <div
      class="return-icon--container"
    >
      <a
        class="return-icon"
        href="/"
      >
        <img
          alt="Retornar à página anterior"
          src="left-arrow.svg"
        />
        Voltar
      </a>
    </div>
  </div>
</div>
`;
